### DISCUSSION TITLE
Electrical bundle routing of ports on the same column

### DISCUSSION THREAD

**Discussion by pl-deassis** [gdsfactory/gdsfactory#2326] [Q&A]
I have a design where some via pads end up aligned on the same column. If I try to use bundle routing to route two ports which have the same X coordinate, but different Y, to adjacent pads, the algoritm ends up placing one route on top of the other for most of the way, until they reach the pads.

Is there any way to work around this? I have tried waypoints, but the documentation is not very clear. Using steps also didn't give me a good solution. The routing ended up creating additional layers and going all over the place.

**Comment by joamatab** [comment]
How about using `gf.routing.route_ports_to_side` first, then `gf.routing.get_bundle`?

**Reply by pl-deassis** [reply]
Thanks for the suggestion. I have just tried it and it didn't work, probably because I have an array with more than one column of ports. I have the feeling that doing it by steps would be the best way, since I could tell the bundle to move in steps relative to the position of each port (from what I understand), but it keeps breaking...

**Reply by pl-deassis** [reply]
To clarify, what I have is essentially an array of via pads with North-South ports which I am routing to larger pads.

**Reply by joamatab** [reply]
It's hard to see the issue without the code to reproduce it:)
